cmd/multigateway/main.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2025 The Multigres Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// multigateway is the top-level proxy that masquerades as a PostgreSQL server,
+// handling client connections and routing queries to multipooler instances.
+package main
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+func setupConfig() {
+	// Define flags
+	pflag.StringP("port", "p", "5432", "Port to listen on")
```

Comment:
We could add a comment here to be explicit that this is the postgres port and the main entry point to interact with the cluster as if it were a postgres server